tests/unit_test/twitter_ads_api/twitter_ads_api_config_test.py

```diff
@@ -2,66 +2,176 @@
+        assert config[0].target.dataset_name == DATASET_NAME
+        assert config[0].target.table_name == TABLE_NAME
 
     def test_should_read_api_secrets(self):
         secrets = {'key1': 'value1', 'key2': 'value2'}
-        config = TwitterAdsApiConfig.from_dict(get_config_for_item_config_dict({
-            **ITEM_CONFIG_DICT,
-            'source': {
-                **SOURCE_CONFIG,
-                'secrets': secrets
-            }
-        }))
-        assert config.source.secrets.mapping == secrets
-
-    def test_should_read_api_secrets_by_key(self):
-        secrets = {'key1': 'value1', 'key2': 'value2'}
-        config = TwitterAdsApiConfig.from_dict(get_config_for_item_config_dict({
-            **ITEM_CONFIG_DICT,
-            'source': {
-                **SOURCE_CONFIG,
-                'secrets': secrets
-            }
-        }))
-        assert config.source.secrets.mapping['key1'] == secrets['key1']
+        config = TwitterAdsApiConfig.parse_config_list_from_dict(
+            get_config_for_item_config_dict(
+                {
+                    **ITEM_CONFIG_DICT_1,
+                    'source': {
+                        **SOURCE_CONFIG_1,
+                        'secrets': secrets
+                    }
+                },
+                {
+                    **ITEM_CONFIG_DICT_WITH_API_QUERY_PARAMETERS,
+                    'source': {
+                        **SOURCE_CONFIG_WITH_API_QUERY_PARAMETERS,
+                        'secrets': secrets
+                    }
+                }
+            )
+        )
+        assert config[0].source.secrets.mapping == secrets
+        assert config[1].source.secrets.mapping == secrets
+
+    def test_should_read_start_time_value_if_defined(self):
+        config = TwitterAdsApiConfig.parse_config_list_from_dict(
+            get_config_for_item_config_dict(
+                ITEM_CONFIG_DICT_1,
+                ITEM_CONFIG_DICT_WITH_API_QUERY_PARAMETERS
+            )
+        )
+        assert config[1].source.api_query_parameters.parameter_values.start_time_value == (
+            START_TIME_VALUE_1
+        )
+        assert config[0].source.api_query_parameters.parameter_values.start_time_value == (
+            ''
+        )
+
+    def test_should_read_from_bigquery_sql_query_and_project_name_if_defined(self):
+        config = TwitterAdsApiConfig.parse_config_list_from_dict(
+            get_config_for_item_config_dict(
+                ITEM_CONFIG_DICT_1,
+                ITEM_CONFIG_DICT_WITH_API_QUERY_PARAMETERS
+            )
+        )
+        assert config[1].source.api_query_parameters.parameter_values.from_bigquery.sql_query == (
+            SQL_QUERY
+        )
+        assert (
+            config[1].source.api_query_parameters.parameter_values.from_bigquery.project_name == (
+                PROJECT_NAME
+            )
+        )
+        assert config[0].source.api_query_parameters.parameter_values.from_bigquery.sql_query == (
```

Comment:
In the case where we haven't defined `api_query_parameters`, I thought the idea was that `api_query_parameters` would then be `None`.
Then one can check `if source_counfig.source.api_query_parameters`.
And if it is is not None, then all required fields within it will be populated.

Reply:
You are right, but think I prefer the check exact value in the fields too.

Reply:
I meant this comment relating to the configuration of `api_query_parameter` to be None, which I think you addressed. Checking individual values is fine.